lib/usdUfe/ufe/UsdAttributeHolder.cpp

```diff
@@ -59,17 +61,58 @@ bool setUsdAttrMetadata(
 
     UsdUfe::AttributeEditRouterContext ctx(attr.GetPrim(), attr.GetName());
 
+    // These metadata have a USD equivalent. Use it to stay compatible with other DCCs:
+    static const auto knownUSDMetadata
+        = std::unordered_set<std::string> { UsdUfe::MetadataTokens->UIDoc,
+                                            UsdUfe::MetadataTokens->UIEnumLabels,
+                                            UsdUfe::MetadataTokens->UIName,
+                                            UsdUfe::MetadataTokens->UIFolder };
+
     PXR_NS::TfToken tok(key);
     if (PXR_NS::UsdShadeNodeGraph(attr.GetPrim())) {
         if (PXR_NS::UsdShadeInput::IsInput(attr)) {
             PXR_NS::UsdShadeInput input(attr);
-            input.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                input.ClearSdrMetadataByKey(tok);
+            } else {
+                input.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
         } else if (PXR_NS::UsdShadeOutput::IsOutput(attr)) {
             PXR_NS::UsdShadeOutput output(attr);
-            output.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                output.ClearSdrMetadataByKey(tok);
+            } else {
+                output.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
+        }
+    }
+
+    if (key == UsdUfe::MetadataTokens->UIDoc) {
+        attr.SetDocumentation(value.get<std::string>());
+        return true;
+    } else if (key == UsdUfe::MetadataTokens->UIEnumLabels) {
+        auto       sdfAttr = TfStatic_cast<SdfAttributeSpecHandle>(attr.GetPropertyStack().front());
+        const auto enumStrings = UsdUfe::splitString(value.get<std::string>(), ", ");
```

Comment:
Is the text format a specification where we know it is a comma-separated string or we just happen to be in-the-know?

Reply:
That is in the MaterialX specification. And since the second argument of UsdUfe::splitString has the "any of" semantics, it will work if there are no spaces in the list of tokens. Though I am unsure if it will behave correctly if there are spaces in the labels themselves: "Blinn, Oren Nayar, Lambert" should unpack as 3 tokens and the space in the middle of "Oren Nayar" needs to be preserved. I will add tests for that.